### ISSUE TITLE
github/directory-file-count broken? "files inaccessible" error.

### ISSUE THREAD

**Issue by rmzelle** [badges/shields#5967]
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [badge-maker NPM package](https://www.npmjs.com/package/badge-maker)

:beetle: **Description**

I'm trying to use the https://img.shields.io/github/directory-file-count badge, but I always seem to get "files inaccessible" as a result. E.g. https://img.shields.io/github/directory-file-count/citation-style-language/locales produces:

live:
<img src="https://img.shields.io/github/directory-file-count/citation-style-language/locales">

screenshot:
![image](https://user-images.githubusercontent.com/77951/102705929-eae07300-425a-11eb-96b4-fab12dfe47f3.png)

:link: **Link to the badge**

https://img.shields.io/github/directory-file-count/citation-style-language/locales

I see the same result for several other uses of this badge on GitHub that I checked: https://github.com/search?q=%22github%2Fdirectory-file-count%22&type=code

:bulb: **Possible Solution**

**Comment by calebcartwright** [comment]
Confirmed. I'm not sure what's going on with this one.

* The issue can be consistently reproduced in production (img.shields.io) with any target, and i've also been unable to get a working badge in prod
* All the other GitHub badges in prod seem fine
* This badge is working fine in other environments, like staging, CI, and local

https://shields-staging.herokuapp.com/github/directory-file-count/citation-style-language/locales
![](https://shields-staging.herokuapp.com/github/directory-file-count/citation-style-language/locales)

**Comment by chris48s** [comment]
I don't know what the issue is here either, but I suspect a key difference between prod and staging here is that staging has a single github token, whereas production uses the token pool.

**Comment by Jsoconno** [comment]
Any word on when this might get fixed?

**Comment by chris48s** [comment]
We don't really know why its failing, so no.

I did have a quick look to check if the contents API is somehow using the search tokens instead of the main pool.. no dice.
Another thing that is super-odd here is that other badges which use the /contents API endpoint work fine e.g:

- https://img.shields.io/github/size/webcaetano/craft/build/phaser-craft.min.js
- https://img.shields.io/nycrc/yargs/yargs

One thing we could do if we're just completely stumped is see if we can rewrite this badge to use a graphql query instead of the V3 API as a workaround :shrug:

**Comment by BenjaminHalko** [comment]
Any update on this problem?

**Comment by calebcartwright** [comment]
As always, the lack of posted updates means that there are no updates.

I'd encourage anyone interested in a resolution to consider contributing by taking a look at implementing the previously proposed solution :point_down: 

> One thing we could do if we're just completely stumped is see if we can rewrite this badge to use a graphql query instead of the V3 API as a workaround shrug

**Comment by pradosh-arduino** [comment]
When can we expect this issue to be fixed? because I really want it!
![Capture](https://user-images.githubusercontent.com/69463173/159723060-0aca8c07-c56d-4384-991b-06bd5c94c7b2.PNG)

can we get a update if this issue was fixed? (in here)

thanks,
Pradosh

**Comment by MarlonAEC** [comment]
Hey I saw there is a PR merged for this issue but I'm still getting the error please see the issue in the image below. Any clues why? Thanks in advance!
<img width="1155" alt="Screen Shot 2022-05-30 at 4 42 16 PM" src="https://user-images.githubusercontent.com/50201790/171015645-8e2003cc-a4e5-49f9-9dd1-58aa3d75580b.png">

**Comment by calebcartwright** [comment]
@MarlonAEC - I'm not sure what PR you're referring to. This issue is still open because the issue still exists. The above commentary covers all known information available at this point.

**Comment by MarlonAEC** [comment]
@calebcartwright yes sorry you are right I thought a PR had already been merged for this issue. My bad thanks!

**Comment by DenverCoder1** [comment]
From testing things out locally, I'm noticing the "inaccessible" error, and it seems like this is the underlying error:
```js
RequestError: `url` must not start with a slash
    at Request._destroy (file:///workspace/shields/node_modules/got/dist/source/core/index.js:458:21)
    at _destroy (node:internal/streams/destroy:102:25)
    at Request.destroy (node:internal/streams/destroy:64:5)
    at Request.flush (file:///workspace/shields/node_modules/got/dist/source/core/index.js:232:22)
    at lastHandler (file:///workspace/shields/node_modules/got/dist/source/create.js:37:26)
    at iterateHandlers (file:///workspace/s…space/shields/core/base-service/base.js:234:40)
    at GithubDirectoryFileCount._requestJson (file:///workspace/shields/core/base-service/base-json.js:45:35)
    at GithubDirectoryFileCount.fetch (file:///workspace/shields/services/github/github-directory-file-count.service.js:121:17)
    at GithubDirectoryFileCount.handle (file:///workspace/shields/services/github/github-directory-file-count.service.js:153:32)
    at Function.invoke (file:///workspace/shields/core/base-service/base.js:407:45)
```
I tried out making a small change to the code to try to test this, and it actually worked.
```diff
  async fetch({ user, repo, path = '' }) {
    return this._requestJson({
      url: `/repos/${user}/${repo}/contents/${path}`,
      url: `https://api.github.com/repos/${user}/${repo}/contents/${path}`,
      schema,
      errorMessages: errorMessagesFor('repo or directory not found'),
    })
  }
```
![image](https://user-images.githubusercontent.com/20955511/173960017-a684defc-d911-4da5-8c55-c20ec3651218.png)

I don't know that this is the ideal solution to the problem, but hopefully this information is helpful.

**Comment by chris48s** [comment]
@DenverCoder1  If you're getting 

```
`url` must not start with a slash
```

I think it is a red herring for this issue. Sounds like your `services.github.baseUri` setting is not set for some reason. It should be picked up from https://github.com/badges/shields/blob/347e3839b87c0f0a118633445b52008224a2dda9/config/default.yml#L19-L21 but maybe you've overridden it in `local.yml`? if you get yourself to whatever state you are in when you're seeing that, I'd guess all of your other GH badges are probably doing the same thing (or at least all the ones that use the V3 API)

**Comment by DenverCoder1** [comment]
Yeah, I hadn't touched the config files at all at first, but I did notice that creating a local.yml solved the issue even without changing the URL. I also can't seem to reproduce the issue anymore for some reason even after removing the local.yml.

In any case, is it known what the actual underlying issue is?

It seems that there are really only 2 possibilities:
1. Failing to send the request in `sendRequest` (`got.js` line 25) - this includes the "url must not start with a slash" possibility but other errors could also occur there
2. It gets handled in `checkErrorResponse` (`check-error-response.js` line 22) for having a response code in the 500's range

Has it been checked to make sure the production environment has the right baseUri and is using it for the file count badges?

As the issue is only in the production environment, it very likely something to do with the config files.

**Comment by DenverCoder1** [comment]
> One thing we could do if we're just completely stumped is see if we can rewrite this badge to use a graphql query instead of the V3 API as a workaround shrug

Assuming this idea works, it should be fixed in #8112

**Comment by chris48s** [comment]
so far so good.. https://img.shields.io/github/directory-file-count/citation-style-language/locales
